src/nupic/algorithms/SpatialPooler.hpp

```diff
@@ -306,8 +308,8 @@ namespace nupic {
            */
           virtual void save(ostream& outStream) const;
 
-          virtual void write(ostream& stream) const;
-          virtual void write(SpatialPoolerProto::Builder& proto) const;
+          using Serializable::write;
```

Comment:
I think that is due to argument dependent lookup. If Serializable is just in namespace `nupic` (not `nupic::base::serializable`) then you may not need this.